Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -459,22 +498,73 @@ internal async Task PerformLeafTypeBehavior(TreeNode treeNode)
             }
         }
 
+        /// <summary>
+        /// Performs Subroutine TreeNodeType behavior.
+        /// </summary>
+        /// <param name="treeNode">The TreeNode containing actions to execute.</param>
+        /// <param name="treeNodeKey">The TreeNode's key where the actions are taking place.</param>
+        /// <exception cref="ArgumentException">If the pre-checks fail.</exception>
+        internal async Task PerformSubroutineTypeBehavior(TreeNode treeNode, string treeNodeKey)
+        {
+            // Perform pre-checks.
+            if (treeNode.Actions == null)
+            {
+                throw new ArgumentException("Subroutine TreeNodeTypes must contain at least one SubroutineAction. TreeNodeKey: " + treeNodeKey);
```

Comment:
>must contain at least one SubroutineAction [](start = 70, length = 42)

suggest "must contain at least one SubroutineAction" => "doesn't contain any action". This is to make the fault message different than the one down below. Easy to tell which condition to trigger the fault.